themes/default/content/docs/concepts/inputs-outputs/_index.md

```diff
@@ -13,30 +13,346 @@ aliases:
 
-- The eventual raw value of the output
-- The dependency on the source(s) of the output value
+```java
+var password = new RandomPassword("password", RandomPasswordArgs.builder()
+    .length(16)
+    .special(true)
+    .overrideSpecial("!#$%&*()-_=+[]{}<>:?")
+    .build());
+
+var example = new Instance("example", InstanceArgs.builder()
+    .instanceClass("db.t3.micro")
+    .allocatedStorage(64)
+    .engine("mysql")
+    .username("someone")
+    .password(password.result()) // We pass the output from password as an input
+    .build());
+```
+
+{{% /choosable %}}
+{{< /chooser >}}
+
+In this case, Pulumi is taking the _output_ from one resource and using it as the _input_ to another resource.
+
+## Outputs
+
+All resources created by Pulumi will have properties which are returned from the cloud provider API. These values are *outputs*.
+
+Outputs are values of type {{< pulumi-output >}}, which behave very much like [promises](https://en.wikipedia.org/wiki/Futures_and_promises) or [monads](https://en.wikipedia.org/wiki/Monad_(functional_programming)). This is necessary because outputs are not fully known until the infrastructure resource has actually completed provisioning, which happens *asynchronously*. Outputs are also how Pulumi tracks dependencies between resources. When an output from one resource has been returned from the cloud provider API, Pulumi can link the two resources together and pass it as the input to another resource.
 
 Pulumi automatically captures dependencies when you pass an output from one resource as an input to another resource. Capturing these dependencies ensures that the physical infrastructure resources are not created or updated until all their dependencies are available and up-to-date.
 
-Because outputs are asynchronous, their actual raw values are not immediately available. If you need to access an output’s raw value—for example, to compute a derived, new value, or because you want to log it—you have these options:
+Because outputs are asynchronous, their actual plain values are not immediately available. If you need to access an output’s plain value—for example, to compute a derived, new value, or because you want to log it—you have these options:
 
-- [Apply](#apply): a callback that receives the raw value, and computes a new output
+- [Apply](#apply): a callback that receives the plain value, and computes a new output
 - [Lifting](#lifting): directly read properties off an output value
 - [Interpolation](#outputs-and-strings): concatenate string outputs with other strings directly
 
 ## Apply
 
-To access the raw value of an output and transform that value into a new value, use {{< pulumi-apply >}}. This method accepts a callback that will be invoked with the raw value, once that value is available.
+To access the _plain_ (or _resolved_) value of an output, use {{< pulumi-apply >}}. This method accepts a callback that will be invoked with the plain value, once that value is available.
```

Comment:
I’m not sure it helps (or is necessary) to introduce “resolved” as well here?

Reply:
Remove all references to "resolved"